client/question/services/ConversationManagerService.js

```diff
@@ -78,9 +108,43 @@ tie.factory('SolutionHandlerService', [
             return CodeRunnerDispatcherService.runCodeAsync(
               language, codeSubmission.getPreprocessedCode()
             ).then(function(preprocessedCodeEvalResult) {
-              feedback = FeedbackGeneratorService.getFeedback(
-                tasks, preprocessedCodeEvalResult,
-                codeSubmission.getRawCodeLineIndexes());
+              var feedbackDetails = computeFeedbackDetails(
+                preprocessedCodeEvalResult);
+
+              // TODO(sll): Remove this case.
+              if (feedbackDetails === null) {
+                feedback = FeedbackGeneratorService.getFeedback(
+                  tasks, preprocessedCodeEvalResult,
+                  codeSubmission.getRawCodeLineIndexes());
+              } else {
+                switch (feedbackDetails.getFeedbackCategory()) {
```

Comment:
oh hey, a switch statement. Haven't seen one of those in years.

Nothing wrong with this; just fun to see.

Reply:
I know, right? I actually had a similar reaction; I generally don't go with switch/case stuff but it seemed to fit really well here.